src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -263,4 +366,47 @@ protected virtual void Dispose(bool disposing)
             Dispose(false);
         }
     }
+
+    internal class ThrowHelper
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+        {
+            if (inputBuffer == null)
+                ThrowArgumentNull(ExceptionArgument.inputBuffer);
+
+            if ((uint)inputCount > inputBuffer.Length)
+                ThrowArgumentOutOfRange(ExceptionArgument.inputCount);
+
+            if (inputOffset < 0)
+                ThrowArgumentOutOfRange(ExceptionArgument.inputOffset);
+
+            if ((inputBuffer.Length - inputCount) < inputOffset)
+                ThrowInvalidOffLen();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Really?

Reply:
It's not necessary -- JIT will inline it even without forcing it.